### PR TITLE
Merge 2.3 master

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

**Comment by hypeJunction** [comment]
a bunch of tests are failing so it will be a painful merge up

**Comment by jdalsem** [comment]
strange... what could have caused it? PHP 5.6 or memcache tests? Any idea?

**Comment by hypeJunction** [comment]
Some API changes.

File: engine/classes/Elgg/BootService.php [hunk 1]
```diff
@@ -220,8 +220,8 @@ private function getBootData(\stdClass $CONFIG, Database $db) {
 	 *
 	 * @return \Stash\Interfaces\ItemInterface
 	 */
-	private function getStashItem(\stdClass $CONFIG) {
-		if (!empty($CONFIG->memcache)) {
+	private function getStashItem(\stdClass $CONFIG, $site_guid) {
```
Comment (jdalsem) [inline engine/classes/Elgg/BootService.php]: this is causing issues... site_guid does not exist anymore

**Comment by jdalsem** [comment]
Replaced by #10570

**Pull request closed without merging**